wire-schema/src/jvmTest/kotlin/com/squareup/wire/schema/PrunerTest.kt

```diff
@@ -1536,4 +1536,148 @@ class PrunerTest {
+        .schema()
+    val pruned = schema.prune(PruningRules.Builder()
+        .include("wire.MessageA")
+        .exclude("wire.*")
+        .build())
+    assertThat(pruned.getType("wire.MessageA")).isNotNull()
+    assertThat(pruned.getType("wire.MessageB")).isNull()
+  }
+
+  @Test @Ignore("Implement #1349")
+  fun fieldIsRetainedIfMorePrecisedRuleExists() {
+    val schema = RepoBuilder()
+        .add("service.proto", """
+            |message MyMessage {
+            |  optional string a = 1;
+            |  optional string b = 2;
+            |}
+            """.trimMargin())
+        .schema()
+    val pruned = schema.prune(PruningRules.Builder()
+        .include("MyMessage#a")
+        .exclude("MyMessage")
+        .build())
+    assertThat(pruned.getType("MyMessage")).isNotNull()
+    val myMessageType = pruned.getType("MyMessage") as MessageType
+    assertThat(myMessageType.field("a")).isNotNull()
+    assertThat(myMessageType.field("b")).isNull()
+  }
+
+  @Test @Ignore("Implement #1349")
+  fun enumConstantIsRetainedIfMorePrecisedRuleExists() {
+    val schema = RepoBuilder()
+        .add("service.proto", """
+            |enum MyEnum {
+            |  A = 1;
+            |  B = 2;
+            |}
+            """.trimMargin())
+        .schema()
+    val pruned = schema.prune(PruningRules.Builder()
+        .include("MyEnum#A")
+        .exclude("MyEnum")
+        .build())
+    assertThat(pruned.getType("MyEnum")).isNotNull()
+    val myEnumType = pruned.getType("MyEnum") as EnumType
+    assertThat(myEnumType.constant("A")).isNotNull()
+    assertThat(myEnumType.constant("B")).isNull()
+  }
+
+  @Test
+  fun excludeTakesPrecedenceIfRulesAreEquals() {
+    val schema = RepoBuilder()
+        .add("service.proto", """
+            |message MyMessage {
+            |  optional string a = 1;
+            |  optional string b = 2;
+            |}
+            """.trimMargin())
+        .schema()
+    val pruned = schema.prune(PruningRules.Builder()
+        .include("MyMessage")
+        .exclude("MyMessage")
+        .build())
+    assertThat(pruned.getType("MyMessage")).isNull()
+  }
+
+  @Test @Ignore("Implement #1349")
+  fun optionFieldIsRetainedIfMorePreciseRuleExists() {
+    val schema = RepoBuilder()
+        .add("lecture.proto", """
+             |package wire;
+             |
+             |import "google/protobuf/descriptor.proto";
+             |
+             |extend google.protobuf.FieldOptions {
+             |  optional bool relevant = 22301;
+             |  optional bool irrelevant = 22302;
+             |  optional bool unused = 22303;
+             |}
+             |
+             |message Lecture {
+             |  optional string title = 1 [(wire.relevant) = true];
+             |  optional string content = 2 [(wire.irrelevant) = true];
+             |}
+             """.trimMargin())
+        .schema()
+    val pruned = schema.prune(PruningRules.Builder()
+        .include("wire.Lecture")
```

Comment:
That's awesome! We want to whitelist parts of `descriptor.proto` we care about and exclude `google.protobuf.*`.

Reply:
I don't think we wanna change our behaviour forbidding the pruning of declared fields inside `descriptor.proto` so the whitelisting of those shouldn't be required.
For Cash Android, this would be something as simple as
```
-google.protobuf.*
+squareup.redacted
```